core/FrontController.php

```diff
@@ -421,8 +421,14 @@ public function init()
         // Force the auth to use the token_auth if specified, so that embed dashboard
         // and all other non widgetized controller methods works fine
         if (Common::getRequestVar('token_auth', '', 'string') !== ''
-            && Request::shouldReloadAuthUsingTokenAuth(null)) {
+            && Request::shouldReloadAuthUsingTokenAuth(null)
+        ) {
             Request::reloadAuthUsingTokenAuth();
+
+            if (Access::getInstance()->hasSuperUserAccess()) {
```

Comment:
could maybe move this into `isTokenAuthLimitedToViewAccess ` method and then rename the method `isTokenAuthLimitedToViewAccess `? Just in case it will ever be reused in other places.

Reply:
Will refactor to merge the logic.